app/components/post/post.js

```diff
@@ -404,30 +403,33 @@ export default class Post extends PureComponent {
         const selected = this.state && this.state.selected ? style.selected : null;
         const highlighted = highlight ? style.highlight : null;
         const isReplyPost = this.isReplyPost();
-
         const onUsernamePress = Config.ExperimentalUsernamePressIsMention ? this.autofillUserMention : this.viewUserProfile;
+        const mergeMessage = isConsecutivePost && !hasComments;
 
         // postWidth = deviceWidth - profilePic width - profilePictureContainer margins - right column margin
         const postWidth = this.props.deviceWidth - 66;
 
         return (
             <View style={[style.container, this.props.style, highlighted, selected]}>
-                <TouchableHighlight
-                    style={[style.profilePictureContainer, (isPostPendingOrFailed(post) && style.pendingPost)]}
-                    onPress={this.handlePress}
-                    onHideUnderlay={this.handleHideUnderlay}
-                    onShowUnderlay={this.handleShowUnderlay}
-                    onLongPress={this.showOptionsContext}
-                    underlayColor='transparent'
-                >
-                    <PostProfilePicture
-                        onViewUserProfile={this.viewUserProfile}
-                        postId={post.id}
-                    />
-                </TouchableHighlight>
+                {mergeMessage ?
```

Comment:
I get what's going on here but it is a little bit difficult to read, perhaps you could do something like

```js
let postHeader;
let userProfile;
if (mergeMessage) {
  userProfile = <View style={style.consecutivePostContainer}/>;
} else {
   userProfile = <TouchableHighlight ...>;
   postHeader = <PostHeader ...>;
}
```

and then just do 
```js
return (
            <View style={[style.container, this.props.style, highlighted, selected]}>
                {userProfile}
                <View style={style.messageContainerWithReplyBar}>
                    {!commentedOnPost && this.renderReplyBar()}
                    <View style={[style.rightColumn, (commentedOnPost && isLastReply && style.rightColumnPadding)]}>
                    {postHeader}
                    ....
```